src/devices/src/virtio/block/device.rs

```diff
@@ -332,886 +332,594 @@ impl VirtioDevice for Block {
+        }
+
+        // Test `get_queue_events()`.
+        {
+            let queue_events = block.get_queue_events().unwrap();
+            assert_eq!(queue_events.len(), 1);
+            let evfd = &queue_events[0];
+            // Check if the two queue eventfds point to the same file
+            // by writing to one and then reading from the other.
+            block.queue_evt.write(1).unwrap();
+            assert!(evfd.read().is_ok());
+        }
+
+        // Test `get_interrupt()`.
+        {
+            let intr = block.get_interrupt().unwrap();
+            // Check if the two interrupt eventfds point to the same file
+            // by writing to one and then reading from the other.
+            block.interrupt_evt.write(1).unwrap();
+            assert!(intr.read().is_ok());
+        }
+
+        // Test `get_interrupt_status()`.
+        {
+            // After initialization, the interrupt status should be zero.
+            assert_eq!(block.get_interrupt_status().load(Ordering::Relaxed), 0);
+        }
+
+        // Test `read_config()`.
+        {
+            let mut num_sectors = [0u8; 4];
+            block.read_config(0, &mut num_sectors);
+            // size is 0x1000, so num_sectors is 8 (4096/512).
+            assert_eq!([0x08, 0x00, 0x00, 0x00], num_sectors);
+            let mut msw_sectors = [0u8; 4];
+            block.read_config(4, &mut msw_sectors);
+            // size is 0x1000, so msw_sectors is 0.
+            assert_eq!([0x00, 0x00, 0x00, 0x00], msw_sectors);
+
+            // Invalid read.
+            num_sectors = [0xd, 0xe, 0xa, 0xd];
+            check_metric_after_block!(
+                &METRICS.block.cfg_fails,
+                1,
+                block.read_config(CONFIG_SPACE_SIZE as u64 + 1, &mut num_sectors)
+            );
+            // Validate read failed.
+            assert_eq!(num_sectors, [0xd, 0xe, 0xa, 0xd]);
+        }
+
+        // Test `features()` and `ack_features()`.
+        {
+            let features: u64 = (1u64 << VIRTIO_BLK_F_RO)
+                | (1u64 << VIRTIO_F_VERSION_1)
+                | (1u64 << VIRTIO_BLK_F_FLUSH);
+
+            assert_eq!(block.avail_features_by_page(0), features as u32);
+            assert_eq!(block.avail_features_by_page(1), (features >> 32) as u32);
+            for i in 2..10 {
+                assert_eq!(block.avail_features_by_page(i), 0u32);
+            }
+
+            for i in 0..10 {
+                block.ack_features_by_page(i, u32::MAX);
+            }
+            assert_eq!(block.acked_features, features);
+        }
+
+        // Test `write_config()`.
+        {
+            let new_config: [u8; 8] = [0x00, 0x50, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00];
+            block.write_config(0, &new_config);
+            let mut new_config_read = [0u8; 8];
+            block.read_config(0, &mut new_config_read);
+            assert_eq!(new_config, new_config_read);
+            // Invalid write.
+            check_metric_after_block!(
```

Comment:
The check for this metric here is not needed. We just need to call `block.write_config` to get the new config writing going, then check that there is no change in the config. WDYT?

Reply:
Agree, removed.